core/state_processor.go

```diff
@@ -151,7 +156,7 @@ func ApplyTransaction(config *params.ChainConfig, bc ChainContext, author *commo
 
 	var cxReceipt *types.CXReceipt
 	if txType == types.SubtractionOnly {
-		cxReceipt = &types.CXReceipt{tx.Hash(), msg.Nonce(), msg.From(), msg.To(), tx.ShardID(), tx.ToShardID(), msg.Value()}
```

Comment:
I forget why we needed this and why we deleted it?

Reply:
the nonce of the receipt is not very useful, we have to sync between source and destination shard's nonce, if only put in one shard of the nonce is not enough. so we use key-value store to track unspent receipt